### ISSUE TITLE
Exception(SerializationInfo info, StreamingContext context) flagged as not compatible

### ISSUE THREAD

**Issue by danielmarbach** [dotnet/platform-compat#66]
```
    [Serializable]
    public class MessageDeserializationException : SerializationException
    {
...

        #pragma warning disable PC001
        protected MessageDeserializationException(SerializationInfo info, StreamingContext context) : base(info, context)
        {
        }
        #pragma warning restore PC001
    }

```

but it seems to be supported?

https://github.com/dotnet/coreclr/blob/release/2.0.0/src/mscorlib/shared/System/Runtime/Serialization/SerializationException.cs#L35

**Comment by bording** [comment]
Looking at this a bit more, the above code isn't enough to trigger a warning, however this does:

```csharp
/// <summary>
/// <see cref="SerializationException(SerializationInfo, StreamingContext)" />
/// </summary>
public class MessageDeserializationException : SerializationException
{
    protected MessageDeserializationException(SerializationInfo info, StreamingContext context) : base(info, context)
    {
    }
}
```

It's actually the usage of `SerializationException` in the xml comment that seems to be causing the warning. As @danielmarbach said, I'm not seeing a reason why this would be flagged, so I think it's a false positive.

**Comment by pjanotti** [comment]
These methods to serialize exceptions were throwing PNSE in the past but eventually the fix was backported to 2.0 (see https://github.com/dotnet/platform-compat/issues/76#issuecomment-350378457).

Anyway, the xml thing here is interesting...

**Comment by OliaG** [comment]
In the scope of this issue, we will fix XML problem and will create a separate issue to redesign for different versions.

### XML problem:
SerializationException causes warning "PC001 SerializationException.SerializationException(SerializationInfo, StreamingContext) isn't supported on Linux, MacOSX, Windows" when it's in XML:
```csharp
/// <summary>
/// <see cref="SerializationException(SerializationInfo, StreamingContext)" />
/// </summary>
```
and does not cause warning when it is in the class body:

```csharp
static void Main(string[] args)
        {
            var e = new SerializationException();
        }
```
